lib/internal/source_map/source_map_cache.js

```diff
@@ -39,12 +40,33 @@ const { fileURLToPath, URL } = require('url');
 let Module;
 let SourceMap;
 
-let experimentalSourceMaps;
-function maybeCacheSourceMap(filename, content, cjsModuleInstance) {
-  if (experimentalSourceMaps === undefined) {
-    experimentalSourceMaps = getOptionValue('--enable-source-maps');
+let _sourceMapsEnabled;
+let _sourceMapOptionChecked;
+function getSourceMapsEnabled() {
+  if (_sourceMapsEnabled) return true;
+  if (_sourceMapOptionChecked === undefined) {
```

Comment:
Unidiomatic variable names (drop the underscores) and I don't think you need `_sourceMapOptionChecked`? You can use `_sourceMapsEnabled` (sans underscore`) as a tri-state: true, false or undefined when it hasn't been set yet.